### PR TITLE
Recursive formatting cmake files with possibility add ignore directories during parsing

### PR DESCRIPTION
**Pull request by dkandyba**
This PR is **ready** for review.
**Summary**
This PR provides new features:

- recursively formatting cmake files;
- possibility add ignore directories when started recursive formatting;

**CLA**
☑︎ I have signed the CLA

### CODE REVIEW

File: cmake_format/__main__.py [hunk 1]
```diff
@@ -167,7 +172,7 @@ def dump_config(args, config_dict, outfile):
 
   for key, value in vars(args).items():
     if (key in configuration.Configuration.get_field_names()
-        and value is not None):
+            and value is not None):
```
Comment (IGapchuk) [inline cmake_format/__main__.py]: @dkandyba As for me it is redundant changes. Please, revert it.

File: cmake_format/__main__.py [hunk 2]
```diff
@@ -302,10 +333,11 @@ def main():
 
     for key, value in vars(args).items():
       if (key in configuration.Configuration.get_field_names()
-          and value is not None):
+              and value is not None):
```
Comment (IGapchuk) [inline cmake_format/__main__.py:341]: @dkandyba As for me it is redundant changes. Please, revert it.

File: cmake_format/__main__.py [hunk 3]
```diff
@@ -325,13 +357,34 @@ def main():
...
+      if ignored:
+        parse_ok = False
+        continue
+
+      if infile_path != '-' and not re.findall(CMAKE_LISTS_RE_PATTERNS, infile_path):
+        sys.stderr.write(
+            "cmake-format error: It is not cmake file: '{0}' \n".format(infile_path))
+        parse_ok = False
+        continue
+
```
Comment (IGapchuk) [inline cmake_format/__main__.py:392]: @dkandyba Please, pay attention to formatting of this for scope. "If" blocks are not alimented.

File: cmake_format/configuration.py [hunk 4]
```diff
@@ -31,6 +31,10 @@ def parse_bool(string):
   return False
 
 
+def parse_as_list(string, sepr=":"):
+  return list(string.split(sepr))
```
Comment (IGapchuk) [inline cmake_format/configuration.py]: @dkandyba What means parameter's name "sepr"? I think will be better "separator", if I have understood right.

File: cmake_format/__main__.py (hunk 3 above)
Comment (dkandyba) [reply cmake_format/__main__.py:392]: @IGapchuk This is normal, that the "if" blocks are in that scope.

File: cmake_format/__main__.py (hunk 1 above)
Comment (dkandyba) [reply cmake_format/__main__.py]: @IGapchuk Fixed in: f860641a48e2dccc2fddd0b5d3962b5cc1b05741

File: cmake_format/__main__.py (hunk 2 above)
Comment (dkandyba) [reply cmake_format/__main__.py:341]: @IGapchuk Fixed in: f860641a48e2dccc2fddd0b5d3962b5cc1b05741

File: cmake_format/configuration.py (hunk 4 above)
Comment (dkandyba) [reply cmake_format/configuration.py]: @IGapchuk Fixed in: f860641a48e2dccc2fddd0b5d3962b5cc1b05741

**Review by IGapchuk — APPROVED** [review]

File: cmake_format/test/cmake-format.json [hunk 5]
```diff
@@ -17,4 +20,4 @@
       }
     }
   }
-}
+}
```
Comment (BSolonenko) [inline cmake_format/test/cmake-format.json]: @dkandyba please add empty line.

**Review by BSolonenko — CHANGES_REQUESTED** [review]
@dkandyba '/path/to/ignore/dir' is in three places. Where do I need to change to add a new directory to the ignore?

File: cmake_format/configuration.py [hunk 6]
```diff
@@ -229,6 +236,8 @@ def linewidth(self):
     "ruler_pattern":
     ("Regular expression to match rulers in comments default=r'{}'"
      .format(DEFAULT_RULER_PATTERN)),
+    "ignore_directories":
+    "list of directories which will be ignore during parsing, use ':' as separator directory",
```
Comment (AByzhynar) [inline cmake_format/configuration.py]: @dkandyba will be ignored

File: cmake_format/__main__.py (hunk 3 above)
Comment (AByzhynar) [reply cmake_format/__main__.py:392]: @dkandyba Make alignment of all blocks inside loop

File: cmake_format/__main__.py [hunk 7]
```diff
@@ -230,6 +245,10 @@ def main():
   mutex.add_argument('-o', '--outfile-path', default=None,
                      help='Where to write the formatted file. '
                           'Default is stdout.')
+
+  mutex.add_argument('-f', '--folder-path', default=None,
+                     help='Folder where should format cmake files.')
```
Comment (AByzhynar) [inline cmake_format/__main__.py]: @dkandyba directory where cmake files should be formatted

**Review by AByzhynar — CHANGES_REQUESTED** [review]

File: cmake_format/test/cmake-format.json (hunk 5 above)
Comment (dkandyba) [reply cmake_format/test/cmake-format.json]: @BSolonenko I think this changes is redundant. What is the reason adding empty line?

File: cmake_format/configuration.py (hunk 6 above)
Comment (dkandyba) [reply cmake_format/configuration.py]: @AByzhynar Fixed in: 87b6fc709025c5bd8eb8be595494c09cbc6207cc

File: cmake_format/__main__.py (hunk 7 above)
Comment (dkandyba) [reply cmake_format/__main__.py]: @AByzhynar  Fixed in: 87b6fc709025c5bd8eb8be595494c09cbc6207cc

File: cmake_format/test/cmake-format.py [hunk 8]
```diff
@@ -13,6 +13,9 @@
 # If true, separate function names from the parenthesis with a space
 separate_fn_name_with_space = False
 
+# list of directories which will be ignore during parsing
```
Comment (AByzhynar) [inline cmake_format/test/cmake-format.py]: @dkandyba `ignore -> ignored`

File: cmake_format/test/cmake-format.yaml [hunk 9]
```diff
@@ -13,6 +13,10 @@ separate_ctrl_name_with_space : false
 # If true, separate function names from the parenthesis with a space
 separate_fn_name_with_space : false
 
+# list of directories which will be ignore during parsing
```
Comment (AByzhynar) [inline cmake_format/test/cmake-format.yaml]: @dkandyba `ignore -> ignored`

**Review by AByzhynar — CHANGES_REQUESTED** [review]

**Comment by AByzhynar** [comment]
@LuxoftAKutsan Please review

File: cmake_format/test/cmake-format.json (hunk 5 above)
Comment (dkandyba) [reply cmake_format/test/cmake-format.json]: @BSolonenko  Fixed in: dc880b1cb71405b990ba4615189e39ccbd14f6a6

File: cmake_format/test/cmake-format.py (hunk 8 above)
Comment (dkandyba) [reply cmake_format/test/cmake-format.py]: @AByzhynar Fixed in: dc880b1cb71405b990ba4615189e39ccbd14f6a6

File: cmake_format/test/cmake-format.yaml (hunk 9 above)
Comment (dkandyba) [reply cmake_format/test/cmake-format.yaml]: @AByzhynar Fixed in: dc880b1cb71405b990ba4615189e39ccbd14f6a6

**Review by LuxoftAKutsan — APPROVED** [review]
Please move all changes to https://github.com/LuxoftSDL/cmake_format

**Pull request merged**